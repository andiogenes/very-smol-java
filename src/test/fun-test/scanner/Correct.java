// Правильный тест (не менее 3 примеров лексем каждого типа)
class/**/class class
void void void
switch switch switch
case case case
default default default
break break break
int int int
short short short
long long long
double double double
return return return//
abcd _______________a1 _abcdefghijklmnopqstxyz0123456789
0 123456/**/405060708090
0.1E5 1050.1E+15 24.1E-15
. . ./*
*/
, , ,
: :/**/:
; ; ;
( ( ( ) ) )
{ { {  } } }
< < < > > >
<= <= <= >= >= >=
== != == != == !=
+ - + - + -
* / * / * /
% = % = % =
&& || && || && ||
++ -- ++ -- ++ --
! ! !