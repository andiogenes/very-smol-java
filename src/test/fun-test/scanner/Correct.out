Type: CLASS, lexeme: class, line: 2, pos: 1
Type: CLASS, lexeme: class, line: 2, pos: 10
Type: CLASS, lexeme: class, line: 2, pos: 16
Type: VOID, lexeme: void, line: 3, pos: 1
Type: VOID, lexeme: void, line: 3, pos: 6
Type: VOID, lexeme: void, line: 3, pos: 11
Type: SWITCH, lexeme: switch, line: 4, pos: 1
Type: SWITCH, lexeme: switch, line: 4, pos: 8
Type: SWITCH, lexeme: switch, line: 4, pos: 15
Type: CASE, lexeme: case, line: 5, pos: 1
Type: CASE, lexeme: case, line: 5, pos: 6
Type: CASE, lexeme: case, line: 5, pos: 11
Type: DEFAULT, lexeme: default, line: 6, pos: 1
Type: DEFAULT, lexeme: default, line: 6, pos: 9
Type: DEFAULT, lexeme: default, line: 6, pos: 17
Type: BREAK, lexeme: break, line: 7, pos: 1
Type: BREAK, lexeme: break, line: 7, pos: 7
Type: BREAK, lexeme: break, line: 7, pos: 13
Type: INT, lexeme: int, line: 8, pos: 1
Type: INT, lexeme: int, line: 8, pos: 5
Type: INT, lexeme: int, line: 8, pos: 9
Type: SHORT, lexeme: short, line: 9, pos: 1
Type: SHORT, lexeme: short, line: 9, pos: 7
Type: SHORT, lexeme: short, line: 9, pos: 13
Type: LONG, lexeme: long, line: 10, pos: 1
Type: LONG, lexeme: long, line: 10, pos: 6
Type: LONG, lexeme: long, line: 10, pos: 11
Type: DOUBLE, lexeme: double, line: 11, pos: 1
Type: DOUBLE, lexeme: double, line: 11, pos: 8
Type: DOUBLE, lexeme: double, line: 11, pos: 15
Type: RETURN, lexeme: return, line: 12, pos: 1
Type: RETURN, lexeme: return, line: 12, pos: 8
Type: RETURN, lexeme: return, line: 12, pos: 15
Type: IDENTIFIER, lexeme: abcd, line: 13, pos: 1
Type: IDENTIFIER, lexeme: _______________a1, line: 13, pos: 6
Type: IDENTIFIER, lexeme: _abcdefghijklmnopqstxyz0123456789, line: 13, pos: 24
Type: NUMBER_INT, lexeme: 0, line: 14, pos: 1
Type: NUMBER_INT, lexeme: 123456, line: 14, pos: 3
Type: NUMBER_INT, lexeme: 405060708090, line: 14, pos: 13
Type: NUMBER_EXP, lexeme: 0.1E5, line: 15, pos: 1
Type: NUMBER_EXP, lexeme: 1050.1E+15, line: 15, pos: 7
Type: NUMBER_EXP, lexeme: 24.1E-15, line: 15, pos: 18
Type: DOT, lexeme: ., line: 16, pos: 1
Type: DOT, lexeme: ., line: 16, pos: 3
Type: DOT, lexeme: ., line: 16, pos: 5
Type: COMMA, lexeme: ,, line: 18, pos: 1
Type: COMMA, lexeme: ,, line: 18, pos: 3
Type: COMMA, lexeme: ,, line: 18, pos: 5
Type: COLON, lexeme: :, line: 19, pos: 1
Type: COLON, lexeme: :, line: 19, pos: 3
Type: COLON, lexeme: :, line: 19, pos: 8
Type: SEMICOLON, lexeme: ;, line: 20, pos: 1
Type: SEMICOLON, lexeme: ;, line: 20, pos: 3
Type: SEMICOLON, lexeme: ;, line: 20, pos: 5
Type: LEFT_PAREN, lexeme: (, line: 21, pos: 1
Type: LEFT_PAREN, lexeme: (, line: 21, pos: 3
Type: LEFT_PAREN, lexeme: (, line: 21, pos: 5
Type: RIGHT_PAREN, lexeme: ), line: 21, pos: 7
Type: RIGHT_PAREN, lexeme: ), line: 21, pos: 9
Type: RIGHT_PAREN, lexeme: ), line: 21, pos: 11
Type: LEFT_BRACE, lexeme: {, line: 22, pos: 1
Type: LEFT_BRACE, lexeme: {, line: 22, pos: 3
Type: LEFT_BRACE, lexeme: {, line: 22, pos: 5
Type: RIGHT_BRACE, lexeme: }, line: 22, pos: 8
Type: RIGHT_BRACE, lexeme: }, line: 22, pos: 10
Type: RIGHT_BRACE, lexeme: }, line: 22, pos: 12
Type: LESS, lexeme: <, line: 23, pos: 1
Type: LESS, lexeme: <, line: 23, pos: 3
Type: LESS, lexeme: <, line: 23, pos: 5
Type: GREATER, lexeme: >, line: 23, pos: 7
Type: GREATER, lexeme: >, line: 23, pos: 9
Type: GREATER, lexeme: >, line: 23, pos: 11
Type: LESS_EQUAL, lexeme: <=, line: 24, pos: 1
Type: LESS_EQUAL, lexeme: <=, line: 24, pos: 4
Type: LESS_EQUAL, lexeme: <=, line: 24, pos: 7
Type: GREATER_EQUAL, lexeme: >=, line: 24, pos: 10
Type: GREATER_EQUAL, lexeme: >=, line: 24, pos: 13
Type: GREATER_EQUAL, lexeme: >=, line: 24, pos: 16
Type: EQUAL_EQUAL, lexeme: ==, line: 25, pos: 1
Type: BANG_EQUAL, lexeme: !=, line: 25, pos: 4
Type: EQUAL_EQUAL, lexeme: ==, line: 25, pos: 7
Type: BANG_EQUAL, lexeme: !=, line: 25, pos: 10
Type: EQUAL_EQUAL, lexeme: ==, line: 25, pos: 13
Type: BANG_EQUAL, lexeme: !=, line: 25, pos: 16
Type: PLUS, lexeme: +, line: 26, pos: 1
Type: MINUS, lexeme: -, line: 26, pos: 3
Type: PLUS, lexeme: +, line: 26, pos: 5
Type: MINUS, lexeme: -, line: 26, pos: 7
Type: PLUS, lexeme: +, line: 26, pos: 9
Type: MINUS, lexeme: -, line: 26, pos: 11
Type: STAR, lexeme: *, line: 27, pos: 1
Type: SLASH, lexeme: /, line: 27, pos: 3
Type: STAR, lexeme: *, line: 27, pos: 5
Type: SLASH, lexeme: /, line: 27, pos: 7
Type: STAR, lexeme: *, line: 27, pos: 9
Type: SLASH, lexeme: /, line: 27, pos: 11
Type: PERCENT, lexeme: %, line: 28, pos: 1
Type: EQUAL, lexeme: =, line: 28, pos: 3
Type: PERCENT, lexeme: %, line: 28, pos: 5
Type: EQUAL, lexeme: =, line: 28, pos: 7
Type: PERCENT, lexeme: %, line: 28, pos: 9
Type: EQUAL, lexeme: =, line: 28, pos: 11
Type: AND, lexeme: &&, line: 29, pos: 1
Type: OR, lexeme: ||, line: 29, pos: 4
Type: AND, lexeme: &&, line: 29, pos: 7
Type: OR, lexeme: ||, line: 29, pos: 10
Type: AND, lexeme: &&, line: 29, pos: 13
Type: OR, lexeme: ||, line: 29, pos: 16
Type: PLUS_PLUS, lexeme: ++, line: 30, pos: 1
Type: MINUS_MINUS, lexeme: --, line: 30, pos: 4
Type: PLUS_PLUS, lexeme: ++, line: 30, pos: 7
Type: MINUS_MINUS, lexeme: --, line: 30, pos: 10
Type: PLUS_PLUS, lexeme: ++, line: 30, pos: 13
Type: MINUS_MINUS, lexeme: --, line: 30, pos: 16
Type: BANG, lexeme: !, line: 31, pos: 1
Type: BANG, lexeme: !, line: 31, pos: 3
Type: BANG, lexeme: !, line: 31, pos: 5
Type: EOF, line: 31, pos: 5